FirebaseSessions/Sources/FirebaseSessions.swift

```diff
@@ -247,18 +247,39 @@ private enum GoogleDataTransportConfig {
     return SessionDetails(sessionId: sessionGenerator.currentSession?.sessionId)
   }
 
+  // This type is not actually sendable, but works around an issue below.
+  // It's safe only if executed on the main actor.
+  private struct MainActorNotificationCallback: @unchecked Sendable {
+    private let callback: (Notification) -> Void
+
+    init(_ callback: @escaping (Notification) -> Void) {
+      self.callback = callback
+    }
+
+    func invoke(notification: Notification) {
+      dispatchPrecondition(condition: .onQueue(.main))
+      callback(notification)
+    }
+  }
+
   func register(subscriber: SessionsSubscriber) {
     Logger
       .logDebug(
         "Registering Sessions SDK subscriber with name: \(subscriber.sessionsSubscriberName), data collection enabled: \(subscriber.isDataCollectionEnabled)"
       )
 
+    // After bumping to iOS 13, this hack should be replaced with `Task { @MainActor in }`
+    let callback = MainActorNotificationCallback { notification in
+      subscriber.onSessionChanged(self.currentSessionDetails)
+    }
+
+    // Guaranteed to execute its callback on the main queue because of the queue parameter.
     notificationCenter.addObserver(
       forName: Sessions.SessionIDChangedNotificationName,
       object: nil,
-      queue: nil
+      queue: OperationQueue.main
     ) { notification in
-      subscriber.onSessionChanged(self.currentSessionDetails)
+      callback.invoke(notification: notification)
```

Comment:
@morganchen12 , do you recall if this change would cause different behavior?

Reply:
Resolving, please comment on this one: https://github.com/firebase/firebase-ios-sdk/pull/14599/files#r2089794215

Reply:
In theory it could, since this notification could be posted from any thread. In practice I think it's only posted once, from the UI thread: https://github.com/firebase/firebase-ios-sdk/blob/1deb75c4cf16217459e0021cb8125a264194c0ef/FirebaseSessions/Sources/FirebaseSessions.swift#L168